settings.gradle.kts

```diff
@@ -1,5 +1,58 @@
+@file:Suppress("UnstableApiUsage")
+
 rootProject.name = "dokka"
 
+pluginManagement {
+    includeBuild("build-logic")
```

Comment:
I moved `includeBuild` into the plugins block [as recommended](https://docs.gradle.org/current/userguide/composite_builds.html#included_plugin_builds)

> You may also use the `includeBuild` mechanism outside `pluginManagement` to include plugin builds. However, this does not support all use cases and including plugin builds like that might be deprecated in a future Gradle version.

I also moved the repository definitions to be above plugins block, since logically the repos should be defined before plugins are resolved - but that's not important and I can revert if you want.

Reply:
I moved it out cause I noticed https://github.com/detekt/detekt/commit/0245b0f24b032649908efa37f78b6abd2374efee.

Reply:
Hm. We don't use configuration cache in this project, do we? So we don't have the warning yet, but it might resurface one day

I don't mind leaving it as is, we can deal with the warning when we get it, but if you do move it out - please add a comment why

Reply:
Make sense.

Reply:
I don't think the warning in https://github.com/detekt/detekt/pull/5690 is about configuration cache, it's about the (configuration of) build cache.

https://github.com/detekt/detekt/blob/b07acae2b5ce2d5c8e8216a4e3e022f89c5fa67f/settings.gradle.kts#L71-L84

But Dokka doesn't have that (JetBrains really should set up a Gradle Build Cache server though!)

Reply:
> JetBrains really should set up a Gradle Build Cache server though!

JFYI we actually have a JetBrains instance for Gradle Enterprise (ge.jetbrains.com), but it didn't look like it using it would benefit us in any way compared to what we have now. If there's some really cool things that can be done with it that will work with Dokka, we can switch to it, no problem

Reply:
If there's public, read-only access to the JetBrains Gradle build cache server then that would definitely help! It means that everyone's builds would be faster.

https://docs.gradle.org/current/userguide/build_cache.html#sec:build_cache_configure_remote